experimental/trusted_shuffler/client/src/main.rs

```diff
@@ -32,27 +35,106 @@ pub struct Opt {
         default_value = "http://localhost:8080"
     )]
     server_url: String,
+    #[structopt(long, help = "The QPS we are aiming to simulate", default_value = "10")]
+    qps: u32,
+    #[structopt(
+        long,
+        help = "How many rounds, i.e., seconds, we sent requests",
+        default_value = "1"
+    )]
+    rounds: u32,
 }
 
-#[tokio::main]
+#[tokio::main(flavor = "multi_thread", worker_threads = 2)]
 async fn main() -> anyhow::Result<()> {
-    env_logger::init();
+    env_logger::builder()
+        .format_timestamp(None)
+        .format_level(false)
+        .format_module_path(false)
+        .format_target(false)
+        .init();
     let opt = Opt::parse();
 
-    let request = "Request 1";
-    info!("Sending request: {}", request);
-    let url = format!("{}/request", &opt.server_url);
-
-    let request_start = Instant::now();
-    let response = send_request(&url, Method::POST, request.as_bytes())
-        .await
-        .context("Couldn't receive response")?;
-    let request_duration = request_start.elapsed();
-
-    let parsed_response = String::from_utf8(response).context("Couldn't decode response body")?;
-    info!(
-        "Received response: {} in {:?}",
-        parsed_response, request_duration
-    );
+    let expected_qps = opt.qps;
```

Comment:
In what sense is this QPS "expected"?

Reply:
See my other comment, IMO this should really be "maximum_qps", i.e. it will limit the actual QPS to be no more than that (but it cannot make it go any faster than the underlying machine can actually go)